src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -766,7 +761,10 @@ public ClusterState execute(ClusterState currentState) {
                         if (updatedState == null) {
                             updatedState = currentState;
                         }
-                        if (shouldIgnoreNewClusterState(logger, currentState, updatedState)) {
+                        if (checkWrongMaster(logger, currentState, updatedState)) {
+                            throw new ElasticsearchIllegalStateException("not following master " + updatedState.nodes().masterNode() + " but following master " + currentState.nodes().masterNode());
```

Comment:
can we use a similar message to the log bellow: "cluster state from a different master then the current one, rejecting (received {}, current {})